cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -561,6 +561,24 @@ func (p *BicepProvider) Deploy(ctx context.Context) (*provisioning.DeployResult,
 		logDS("%s", err.Error())
 	}
 
+	deploymentTags := map[string]*string{
+		azure.TagKeyAzdEnvName: to.Ptr(p.env.Name()),
+	}
+	if parametersHashErr == nil {
+		deploymentTags[azure.TagKeyAzdDeploymentStateParamHashName] = to.Ptr(currentParamsHash)
+	}
+
+	err = p.validatePreflight(
+		ctx,
+		bicepDeploymentData.Target,
+		bicepDeploymentData.CompiledBicep.RawArmTemplate,
+		bicepDeploymentData.CompiledBicep.Parameters,
+		deploymentTags,
+	)
```

Comment:
I think it is great that the validate API exists but I am concerned that this may cause considerable time delay during deployments.

Do we have an idea of the amount of extra time the validations will take especially against more complex & nested deployments?

Reply:
@hemarina and I had spoke awhile back about this while she was implementing -- I think this is pretty lightweight (<3s) from what she showed me, but it'd be great to test this out and gather the numbers with a few different templates.